java/vector/src/main/codegen/templates/UnionMapWriter.java

```diff
@@ -98,6 +98,18 @@ public UnionMapWriter(MapVector vector) {
     entryWriter = struct();
   }
 
+  @Override
+  public void start() {
+    writer.setAddVectorAsNullable(false);
```

Comment:
shouldn't this depend on mode?

Reply:
I think this is corresponding to MapVector’s null info, for example
set false in start() means its inner struct vector is non-nullable(in normal usage, this should set in startEntry, but in ComplexCopier startEntry was not invoked)